### PR TITLE
Transaction forwarding limit

### PR DESCRIPTION
**Pull request by sakridge**
#### Problem

Transaction forwarding has no limit

#### Summary of Changes

Put a limit on forwarded packets

Fixes #

### CODE REVIEW

File: core/src/banking_stage.rs [hunk 1]
```diff
@@ -360,11 +364,21 @@ impl BankingStage {
...
+        const INTERVAL_MS: u64 = 100;
+        const MAX_BYTES_PER_SECOND: usize = 10_000 * 1200;
+        const MAX_BYTES_PER_INTERVAL: usize = MAX_BYTES_PER_SECOND * INTERVAL_MS as usize / 1000;
+        const MAX_BYTES_BUDGET: usize = MAX_BYTES_PER_INTERVAL * 5;
+        data_budget.update(INTERVAL_MS, |bytes| {
+            std::cmp::min(bytes + MAX_BYTES_PER_INTERVAL, MAX_BYTES_BUDGET)
+        });
         for p in packets {
-            socket.send_to(&p.data[..p.meta.size], &tpu_forwards)?;
+            if data_budget.take(p.meta.size) {
```
Comment (jstarry) [inline core/src/banking_stage.rs:379]: I think I'd like to rebase https://github.com/solana-labs/solana/pull/19611 off of this PR and apply the data budget before sorting by weight fyi. Could be worth applying this data budget as part of the `filter_valid_packets_for_forwarding` filter with that in mind.

Comment (jstarry) [reply core/src/banking_stage.rs:379]: ah nvm, missed that this change was already merged into v1.6 as https://github.com/solana-labs/solana/pull/19834

Comment (sakridge) [reply core/src/banking_stage.rs:379]: Yea, I was trying to keep the branches consistent.

Comment (sakridge) [reply core/src/banking_stage.rs:379]: We can make a follow-up change that applies on top with the changes you suggest.

**Comment by jstarry** [comment]
@sakridge do you still want to get this and https://github.com/solana-labs/solana/pull/20001 into v1.7 soon?

**Comment by sakridge** [comment]
> @sakridge do you still want to get this and #20001 into v1.7 soon?

yea

**Pull request merged**